### ISSUE TITLE
Improving documentation style

### ISSUE THREAD

**Issue by zar1** [dssg/UPSG#12]
The way we do docstrings is modeled after sklearn (e.g. https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/svm/classes.py). But there's a few stylistic things I've been lax about:
1) Section headings should be capitalized, so

```
Parameters
----------

not

parameters
----------
```

2) Entries in the Parameters section should be written with a space between the variable name and the colon

```
NAME : TYPE

not

NAME: TYPE
```

3) Parameters for **init** should be in the class docstring, not the **init** docstring

4) Public attributes should be documented in the class's Attributes section

5) All classes inheriting from Stage should have "Input Keys" and "Output Keys" sections with format

```
Input Keys
----------
KEY_1_NAME 
    DESCRIPTION

KEY_2_NAME
    DESCRIPTION

etc.
```

6) sklearn gives the returned variable a name, but I think that's silly. It should just be:

```
Returns
-------
    DESCRIPTION_INCLUDING_TYPE
```

**Comment by zar1** [comment]
SKlearn adheres to the numpy docstring standard (https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) so we will too

**Comment by zar1** [comment]
Fixed in b80d7bd3ea
@zar1 zar1 authored 14 seconds ago 
(see issue #44)
